src/main/java/org/apache/maven/plugins/dependency/ListClassesMojo.java

```diff
@@ -0,0 +1,391 @@
+     * central::default::https://repo.maven.apache.org/maven2,myrepo::::https://repo.acme.com,https://repo.acme2.com
+     */
+    @Parameter( property = "remoteRepositories" )
+    private String remoteRepositories;
+
+    /**
+     * A string of the form groupId:artifactId:version[:packaging[:classifier]].
+     */
+    @Parameter( property = "artifact" )
+    private String artifact;
+
+    @Parameter( defaultValue = "${project.remoteArtifactRepositories}", readonly = true, required = true )
+    private List<ArtifactRepository> pomRemoteRepositories;
+
+    /**
+     * Download transitively, retrieving the specified artifact and all of its dependencies.
+     */
+    @Parameter( property = "transitive", defaultValue = "false" )
+    private boolean transitive = false;
+
+    /**
+     * Skip plugin execution completely.
+     *
+     */
+    @Parameter( property = "mdep.skip", defaultValue = "false" )
+    private boolean skip;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException
+    {
+        ProjectBuildingRequest buildingRequest = makeBuildingRequest();
+
+        try
+        {
+            if ( isTransitive() )
+            {
+                Iterable<ArtifactResult> artifacts = dependencyResolver
+                        .resolveDependencies( buildingRequest, coordinate, null );
+
+                for ( ArtifactResult result : artifacts )
+                {
+                    printClassesFromArtifactResult( result );
+                }
+            }
+            else
+            {
+                ArtifactResult result = artifactResolver
+                        .resolveArtifact( buildingRequest, toArtifactCoordinate( coordinate ) );
+
+                printClassesFromArtifactResult( result );
+            }
+        }
+        catch ( ArtifactResolverException | DependencyResolverException | IOException e )
+        {
+            throw new MojoExecutionException( "Couldn't download artifact " + artifact + ": " + e.getMessage(), e );
+        }
+    }
+
+    private void printClassesFromArtifactResult( ArtifactResult result )
+            throws IOException
+    {
+        JarFile jarFile = new JarFile( result.getArtifact().getFile() );
+        Enumeration entries = jarFile.entries();
+
+        while ( entries.hasMoreElements() )
+        {
+            JarEntry entry = (JarEntry) entries.nextElement();
+            String entryName = entry.getName();
+
+            // filter out files that do not end in .class
+            if ( !entryName.endsWith( ".class" ) )
+            {
+                continue;
+            }
+
+            // remove .class from the end and change format to use periods instead of forward slashes
+            String className = entryName.substring( 0, entryName.length() - 6 ).replace( '/', '.' );
+            getLog().info( className );
+        }
+        jarFile.close();
```

Comment:
use try-with-resources to guarantee that this is closed when an exception is thrown

Reply:
fixed